docs/reference/indices/component-templates.asciidoc

```diff
@@ -0,0 +1,272 @@
+[[component-templates]]
+=== Component template API
+++++
+<titleabbrev>Component template</titleabbrev>
+++++
+
+Component templates are building blocks that specify mappings, settings, or alias configuration, but
+don't apply to a set of indices themselves. To be used a component template must be specified in the
+`composed_of` of an <<index-templates,index template>>.
+
+[source,console]
+--------------------------------------------------
+PUT _component_template/template_1
+{
+  "template": {
+    "settings": {
+      "number_of_shards": 1
+    },
+    "mappings": {
+      "_source": {
+        "enabled": false
+      },
+      "properties": {
+        "host_name": {
+          "type": "keyword"
+        },
+        "created_at": {
+          "type": "date",
+          "format": "EEE MMM dd HH:mm:ss Z yyyy"
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TESTSETUP
+
+//////////////////////////
+
+[source,console]
+--------------------------------------------------
+DELETE _component_template/template_*
+--------------------------------------------------
+// TEARDOWN
+
+//////////////////////////
+
+[[put-component-template-api-request]]
+==== {api-request-title}
+
+`PUT /_component_template/<component-template>`
+
+
+[[put-component-template-api-desc]]
+==== {api-description-title}
+
+Use the PUT component template API to create or update a component template.
+
+// tag::component-template-def[]
+Component templates define <<index-modules-settings,settings>>, <<mapping,mappings>>, and
+<<indices-aliases,aliases>> that you can automatically apply when creating new indices. On their own
+component templates are not matched or applied to an index, instead they are used in the
+`composed_of` definition of an <<index-templates-v2,index template>> to make up the final index
+configuration.
+// end::component-template-def[]
+
+Component templates are only used during index creation.
+Changes to component templates do not affect existing indices.
+Settings and mappings specified in <<indices-create-index, create index>> API requests
+override any settings or mappings specified in a component template.
```

Comment:
Should we mention this in the index template (`<<index-templates-v2,index templates>>`) documentation? (as component templates on their own don't interact with indices in any way)

Reply:
I do have a mention of this at the top of the index template docs, but I can make it a bit clearer.